Sources/StreamChat/Workers/ChannelMemberUpdater.swift

```diff
@@ -38,6 +38,33 @@ class ChannelMemberUpdater: Worker {
             }
         }
     }
+    
+    func pinMemberChannel(
+        _ isPinned: Bool,
+        userId: UserId,
+        cid: ChannelId,
+        completion: @escaping (Error?) -> Void
+    ) {
+        partialUpdate(
+            userId: userId,
+            in: cid,
+            extraData: isPinned ? ["pinned": .bool(true)] : nil,
```

Comment:
The overall implementation LGTM, only this part I have some doubts. The "pinned" should be a native field on our side right? So it should not be extra data?

Reply:
I'll make it more explicit.

Reply:
I reviewed other partial methods we have. I think the naming `extraData` is a bit unfortunate here. It is actually `set`.

```
static func partialMemberUpdate(
        userId: UserId,
        cid: ChannelId,
        extraData: [String: RawJSON]?,
        unset: [String]?
    ) -> Endpoint<PartialMemberUpdateResponse> {
        var body: [String: AnyEncodable] = [:]
        if let extraData {
            body["set"] = AnyEncodable(extraData)
        }
        if let unset {
            body["unset"] = AnyEncodable(unset)
        }
```
It actually does not mean all the data is going to be part of member's extra data. All in all, it is correct.

Channel updater has

```
func partialChannelUpdate(
        updates: ChannelEditDetailPayload, // <--- to `set`
        unsetProperties: [String], <-- `to unset`
        completion: ((Error?) -> Void)? = nil
    ) {
```

@nuno-vieira I am not sure if I should change anything here at the moment. Ideally we should fix this naming in the next major. Meanwhile, I could add a new partial update method with better naming, but does it really make sense at the moment.

Reply:
The damage for this one is already done long time ago. It should be `set`, not `extraData`, but there's no point in changing it only here. Imo, we should check if we have it in the tech debt / v5 docs.

Reply:
@martinmitrevski This is a new method, and it was actually introduced by me. Initially, the goal was just to update extra data, which is it is named like that. The backend structure, also make is kinda hard to map this properly. But yeah, the `CHannelEditPayload` is what we want in this case actually. So I would prefer we do something like that. So `ChannelMemberUpdatePayload`